R/read_txt_Horiba.R

```diff
@@ -87,3 +87,59 @@ read_txt_Horiba_t <- function(file, header = TRUE, sep = "\t", row.names = NULL,
 # Unit tests -----------------------------------------------------------------
 
 # FIXME: add unit tests
+
+hySpc.testthat::test(read_txt_Horiba_t) <- function() {
+  context("read_txt_Horiba_t")
+
+  ts <- system.file("extdata/txt.HoribaJobinYvon/", "ts.txt",package = "hySpc.read.txt")
+  spc <- read_txt_Horiba_t(ts)
+
+  test_that("Horiba time series .txt labels are correct",{
+    expect_true(is.expression(spc@label$.wavelength))
+    expect_true(is.expression(spc@label$spc))
+    expect_equal(spc@label$filename, "filename")
+  })
+
+  test_that("Horiba time series .txt spectral data", {
+    expect_equal(dim(spc@data$spc), c(100,1024))
+
+    expect_equal(colnames(spc@data$spc)[[974]], "5.9526")
+    expect_equal(colnames(spc@data$spc)[[657]], "772.27")
+  })
+
+  test_that("Horiba time series .txt wavelength", {
+    expect_equal(length(spc@wavelength), 1024)
+
+    expect_equal(round(spc@wavelength[[79]], 3), 1986.863)
+    expect_equal(round(spc@wavelength[[1011]], 3), -88.828)
+  })
+}
+
+hySpc.testthat::test(read_txt_Horiba_xy) <- function() {
+  context("read_txt_Horiba_t")
+
+  map <- system.file("extdata/txt.HoribaJobinYvon/", "map.txt",package = "hySpc.read.txt")
+  spc <- read_txt_Horiba_xy(map)
+
+  test_that("Horiba map .txt labels are correct",{
+    expect_true(is.expression(spc@label$.wavelength))
+    expect_true(is.expression(spc@label$spc))
+    expect_equal(spc@label$filename, "filename")
+  })
+
+  test_that("Horiba map .txt spectral data", {
+    expect_equal(dim(spc@data$spc), c(141,616))
+
+    expect_equal(colnames(spc@data$spc)[[231]], "1340.24")
+    expect_equal(colnames(spc@data$spc)[[123]], "734.39")
+  })
+
+  test_that("Horiba map .txt wavelength", {
+    expect_equal(length(spc@wavelength), 616)
+
+    expect_equal(round(spc@wavelength[[85]], 3), 521.219)
+    expect_equal(round(spc@wavelength[[431]], 3), 2462.195)
+  })
+}
```

Comment:
please also test the map dimensions x and y

Reply:
Add this as additional unit tests